### PR TITLE
EL-495 add partner dependants section into journey

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/EL-494

Add partner dependants screens to journey. 
Add partner dependants to CFE partner endpoint

### CODE REVIEW

File: app/services/check_answers/applicant_section.rb [hunk 1]
```diff
@@ -18,6 +23,11 @@ class ApplicantSection
     attribute :dependants, :boolean
     attribute :child_dependants, :integer
     attribute :adult_dependants, :integer
+    attribute :partner_employed, :boolean
```
Comment (patrick-laa) [inline app/services/check_answers/applicant_section.rb:26]: I'm not sure why we're adding these attributes here? None of these fields belong in the applicant section of check answers, so I don't think they should belong in CheckAnswers::ApplicantSection.

**Review by patrick-laa — APPROVED** [review]

File: app/lib/check_answers_fields.yml [hunk 2]
```diff
@@ -45,13 +45,16 @@ sections:
           type: boolean
         - attribute: passporting
           type: boolean
-        - attribute: dependants
-          type: boolean
-          screen: dependants
         - attribute: child_dependants
+          type: boolean
+          screen: dependant_details
+        - attribute: child_dependants_count
```
Comment (patrick-laa) [inline app/lib/check_answers_fields.yml:51]: If I say I have 2 child dependants, then go back and change my answer about child dependants from 'Yes' to 'No', the check answers screen will say "**Has child dependants** No"/"**Number of child dependants** 2" which seems wrong. I'd suggest adding a `skip_unless: child_dependants` to this attribute (and similar to the other three `_dependants_count` attributes.

Comment (starswan) [reply app/lib/check_answers_fields.yml:51]: It was done for partner version, just not for client version

File: app/lib/check_answers_fields.yml [hunk 3]
```diff
@@ -45,13 +45,16 @@ sections:
           type: boolean
         - attribute: passporting
           type: boolean
-        - attribute: dependants
-          type: boolean
-          screen: dependants
         - attribute: child_dependants
```
[the remark this answers is not in the document]
Comment (starswan) [reply app/lib/check_answers_fields.yml:51]: Spoken to Kate - she agreed so have implemented

**Pull request merged**